sklearn/utils/__init__.py

```diff
@@ -590,3 +591,84 @@ def is_scalar_nan(x):
     # Redondant np.floating is needed because numbers can't match np.float32
     # in python 2.
     return bool(isinstance(x, (numbers.Real, np.floating)) and np.isnan(x))
+
+def safe_name_or_class(obj):
+    """Return a representation of the name or class, or raise AttributeError.
+    
+    If the object has a non-None __name__, its safe_str() is returned.
+    Otherwise something similar is done with the object's __class__.
+    Otherwise AttributeError is raised.
+    """
+
+    name = getattr(obj, '__name__', None)
+    if name is not None:
+        return safe_str(name)
+
+    klass = getattr(obj, '__class__', None)
+    if klass is not None:
+        return '{} instance'.format(safe_str(klass))
+    else:
+        raise AttributeError("No __name__ or __class__ attribute.")
+
+def safe_version(transformers, no_transform):
+    """Return a function that applies the first working callable.
```

Comment:
"The first working callable" => "The first callable that does not raise an error".

Reply:
I wanted to do this, but then we go beyond the 79-character limit of PEP 8.

I am thus changing this to "the first callable that returns a value".

The details should be clear in the docstring, in any case.

Reply:
Addressed in https://github.com/lebigot/scikit-learn/commit/6bc2a39fe5e6a9f65c9d54fb442e18a265e82e9a.